include/mbedtls/mbedtls_config.h

```diff
@@ -1241,8 +1241,7 @@
  * If you enable this option, you do not need to enable any ECC-related
```

Comment:
There are many places in this file where we list a legacy option and the corresponding PSA option, e.g.

```
 * Requires: MBEDTLS_ECDH_C or PSA_WANT_ALG_ECDH
 *           MBEDTLS_ECDSA_C or PSA_WANT_ALG_ECDSA
 *           MBEDTLS_X509_CRT_PARSE_C
```

Part of the retirement of the legacy configuration mechanism is removing the ability for users to specify “I want ECDSA” by `#define MBEDTLS_ECDSA_C`. Strictly speaking, this isn't directly related to `MBEDTLS_PSA_CRYPTO_CONFIG`, since even with `MBEDTLS_PSA_CRYPTO_CONFIG` enabled, in Mbed TLS 3, putting `#define MBEDTLS_ECDSA_C` in your configuration is valid and means “I want the `ecdsa.h` API to work”. Whether that enables ECDSA through PSA depends on `MBEDTLS_PSA_CRYPTO_CONFIG`. Whether that is good enough for places in PK/X.509/TLS1.2 that require ECDSA depends on of `MBEDTLS_USE_PSA_CRYPTO`.

I don't think this pull request should try to fix these statements, both because it's already pretty long and because it would conflict with https://github.com/Mbed-TLS/mbedtls/pull/9781 (which is what got me to think about this). I suggest planning a follow-up task to clean up requirement statements in the config file(s) once the dust has settled, but before the 4.0 release.

Reply:
Fixing these statements could be done as part of removing the legacy configuration options like MBEDTLS_ECDH_C and MBEDTLS_ECDSA_C: symbols that are in the "Legacy cryptography" section in config-split.md. Wdyt?